### PR TITLE
Validation of rengo auto-start

### PR DESCRIPTION
**Pull request by GreenAsJade**
Fixes tricksy people trying to see what happens if illegal values of auto_start are used

## Proposed Changes

Validate so they can't enter illegal values.

Also, make the display of "no autostart" be "blank" not "0", lest someone raise the concern that "hey, this is going to auto start immediately!?"

### CODE REVIEW

File: src/components/ChallengeModal/ChallengeModal.tsx [hunk 1]
```diff
@@ -700,9 +700,20 @@ export class ChallengeModal extends Modal<Events, ChallengeModalProperties, any>
...
     update_rengo_auto_start = (ev) => {
-        // '0' means "none", but people may like to see it empty.
-        if (ev.target.value === "" || !isNaN(parseInt(ev.target.value))) {
-            this.upstate("challenge.rengo_auto_start", ev);
+        let new_val = parseInt(ev.target.value);
+        if (isNaN(new_val)) {
+            new_val = 0;
+        }
+        // skip over illegal values 1 and 2
+        if (!this.state.challenge.rengo_auto_start && new_val && new_val < 3) {
```
[the remark this answers is not in the document]
Comment (GreenAsJade) [reply src/components/ChallengeModal/ChallengeModal.tsx:708]: LOL good catch 🤦‍♀️

[the remark this answers is not in the document]
Comment (GreenAsJade) [reply src/components/ChallengeModal/ChallengeModal.tsx:708]: I did the onBlur thing, but I'm not wild about it.

The problem is that by then you don't know which direction they were going if they were using the arrows, and you can blur out of the <input> control by clicking "Submit".   The end result being you create a challenge without noticing that it has auto-start not the way you set it.

I just thought of a different way: disable Submit while the value is '1' or '2'... I will see how that goes

**Comment by GreenAsJade** [comment]
Yeah, I think that's a better user experience.

File: src/components/ChallengeModal/ChallengeModal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (anoek) [reply src/components/ChallengeModal/ChallengeModal.tsx]: I like it

**Pull request merged**